### ISSUE TITLE
Error formatting message: "validation.required" for locale (React Intl)

### ISSUE THREAD

**Issue by MatheusRV** [getferdi/ferdi#340]
**Describe the bug**
Unable to Create Account on Ferdi Server

**To Reproduce**

1. Set Ferdi api server
2. Go to 'Criar uma conta gratuita'
3. Click 'Criar uma conta'

**Expected behavior**
Normal account created

**Desktop:**
- OS: Windows 10 19551
- Browser Chrome
- Version 5.4.4-beta.1 e 5.3.3

**Additional context**
```
Failed to load resource: the server responded with a status of 401 ()
api.getferdi.com/v1/…ents/5.4.4-beta.1:1 Failed to load resource: the server responded with a status of 404 ()
api.getferdi.com/v1/me/services:1 Failed to load resource: the server responded with a status of 401 ()
C:\Users\mathe\AppDa…t\instrument.js:104 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in WindowControls
    in div
    in Unknown (created by _class2)
    in div (created by _class2)
    in _class2 (created by Jss(_class2))
    in Jss(_class2) (created by _class2)
    in _class2 (created by Jss(_class2))
    in Jss(_class2) (created by _class2)
api.getferdi.com/v1/me/services:1 Failed to load resource: the server responded with a status of 401 ()
api.getferdi.com/v1/me:1 Failed to load resource: the server responded with a status of 401 ()
7
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.required" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} obrigatório'
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.minLength" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} deve ter pelo menos {comprimento} mais caracteres '
C:\Users\mathe\AppDa…h\SignupScreen.js:1 Uncaught (in promise) TypeError: Cannot read property 'currencyID' of undefined
    at _class2.onSignup (C:\Users\mathe\AppDa…h\SignupScreen.js:1)
    at Object.onSubmit (C:\Users\mathe\AppDa…h\SignupScreen.js:1)
    at DefaultForm.onSuccess (C:\Users\mathe\AppDa…ts\auth\Signup.js:1)
    at DefaultForm.Base.execHook (C:\Users\mathe\AppDa…form\lib\Base.js:67)
    at exec (C:\Users\mathe\AppDa…ared\Actions.js:133)
    at C:\Users\mathe\AppDa…ared\Actions.js:142
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.required" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} obrigatório'
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.minLength" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} deve ter pelo menos {comprimento} mais caracteres '
4
C:\Users\mathe\AppDa…\dist\helpers.js:91 Uncaught SyntaxError: C:\Users\mathe\Downloads\FerdiAppData\window-state.json: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at readFileSync (C:\Users\mathe\AppDa…sonfile\index.js:63)
    at _temp.get debugInfo (C:\Users\mathe\AppDa…tores\AppStore.js:1)
    at ComputedValue.computeValue (C:\Users\mathe\AppDa…bx\lib\mobx.js:1258)
    at ComputedValue.get (C:\Users\mathe\AppDa…bx\lib\mobx.js:1188)
    at ObservableObjectAdministration.read (C:\Users\mathe\AppDa…bx\lib\mobx.js:3992)
    at _temp.get (C:\Users\mathe\AppDa…bx\lib\mobx.js:4256)
    at Proxy.click (C:\Users\mathe\AppDa….asar\lib\Menu.js:1)
    at MenuListItem.handleClick (C:\Users\mathe\AppDa…ebar\lib\menu.js:73)
    at HTMLUnknownElement.callCallback (C:\Users\mathe\AppDa….development.js:336)
    at HTMLUnknownElement.sentryWrapped (C:\Users\mathe\AppDa…\dist\helpers.js:74)
    at Object.invokeGuardedCallbackDev (C:\Users\mathe\AppDa….development.js:385)
    at invokeGuardedCallback (C:\Users\mathe\AppDa….development.js:440)
    at invokeGuardedCallbackAndCatchFirstError (C:\Users\mathe\AppDa….development.js:454)
    at executeDispatch (C:\Users\mathe\AppDa….development.js:584)
    at executeDispatchesInOrder (C:\Users\mathe\AppDa….development.js:606)
    at executeDispatchesAndRelease (C:\Users\mathe\AppDa….development.js:713)
    at executeDispatchesAndReleaseTopLevel (C:\Users\mathe\AppDa….development.js:722)
    at forEachAccumulated (C:\Users\mathe\AppDa….development.js:694)
    at runEventsInBatch (C:\Users\mathe\AppDa….development.js:739)
    at runExtractedPluginEventsInBatch (C:\Users\mathe\AppDa….development.js:880)
    at handleTopLevel (C:\Users\mathe\AppDa…development.js:5803)
    at batchedEventUpdates$1 (C:\Users\mathe\AppDa…evelopment.js:24401)
    at batchedEventUpdates (C:\Users\mathe\AppDa…development.js:1415)
    at dispatchEventForPluginEventSystem (C:\Users\mathe\AppDa…development.js:5894)
    at attemptToDispatchEvent (C:\Users\mathe\AppDa…development.js:6010)
    at dispatchEvent (C:\Users\mathe\AppDa…development.js:5914)
    at unstable_runWithPriority (C:\Users\mathe\AppDa….development.js:697)
    at runWithPriority$2 (C:\Users\mathe\AppDa…evelopment.js:12149)
    at discreteUpdates$1 (C:\Users\mathe\AppDa…evelopment.js:24417)
    at discreteUpdates (C:\Users\mathe\AppDa…development.js:1438)
    at dispatchDiscreteEvent (C:\Users\mathe\AppDa…development.js:5881)
    at HTMLDocument.sentryWrapped (C:\Users\mathe\AppDa…\dist\helpers.js:74)

```

**Comment by kytwb** [comment]
I note:

```
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.required" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} obrigatório'
C:\Users\mathe\AppDa…t\instrument.js:104 [React Intl] Error formatting message: "validation.minLength" for locale: "pt-BR", using default message as fallback.
Error: The intl string context variable 'campo' was not provided to the string '{campo} deve ter pelo menos {comprimento} mais caracteres '
C:\Users\mathe\AppDa…h\SignupScreen.js:1 Uncaught (in promise) TypeError: Cannot read property 'currencyID' of undefined
    at _class2.onSignup (C:\Users\mathe\AppDa…h\SignupScreen.js:1)
    at Object.onSubmit (C:\Users\mathe\AppDa…h\SignupScreen.js:1)
    at DefaultForm.onSuccess (C:\Users\mathe\AppDa…ts\auth\Signup.js:1)
    at DefaultForm.Base.execHook (C:\Users\mathe\AppDa…form\lib\Base.js:67)
    at exec (C:\Users\mathe\AppDa…ared\Actions.js:133)
    at C:\Users\mathe\AppDa…ared\Actions.js:142
```

Looks like an error with the `pt-BR` translation file cause by https://github.com/getferdi/ferdi/blob/f74636d1238128096b11044e181831142ca1c887/src/i18n/locales/pt-BR.json#L480

**Comment by kytwb** [comment]
Getting fixed via https://github.com/getferdi/ferdi/pull/391.

**Comment by kytwb** [comment]
<img width="1226" alt="Screen Shot 2020-02-21 at 10 47 06 PM" src="https://user-images.githubusercontent.com/412895/75074500-1dff8c00-54fc-11ea-850c-8fda0da29b56.png">

**Comment by kytwb** [comment]
@vantezzen I am not sure why some strings are out of sync with Crowdin, see: 

https://github.com/getferdi/ferdi/blob/f74636d1238128096b11044e181831142ca1c887/src/i18n/locales/pt-BR.json#L480

**Comment by kytwb** [comment]
Aside from this, I'm surprised it throws an error instead of graciously failing on fallback/default message. We might want to take a look at this and consider upgrading by the way.

**Comment by kytwb** [comment]
@MatheusRV Thank you for the bug report 🙏
Until we fix the issue, you could try using Ferdi wth the English language if the issue is still relevant to you.

**Comment by kytwb** [comment]
Files affected where variables have been translated as well:

<img width="431" alt="Screen Shot 2020-02-22 at 2 31 04 AM" src="https://user-images.githubusercontent.com/412895/75083743-6c705300-551b-11ea-9a6e-26a5ad781623.png">

**Comment by kytwb** [comment]
@vantezzen Is it ok if I replace in the files or should I first replace in Crowdin and replace the remaining manually on the repository? It seems some are out of sync with Crowdin.

**Comment by vantezzen** [comment]
In my experiences, CrowdIn will completely overwrite all changes you make manually, so you'll have to fix this in CrowdIn.

**Comment by kytwb** [comment]
@vantezzen I checked in Crowdin, but some translations are out of sync with what's in the repository (the variable names are correct in Crowdin but not in the repository files). I will double check everything in Crowdin and manually override the ones out of sync in the repository; I will upgrade `react-intl` at the same time to make sure we'r up to date with it and double check so that we don't throw/crash in these cases (variables translated thus missing in the key) as they could easily slip in.

**Comment by kytwb** [comment]
Should be fixed in the next release. 

This would require to be careful not to re-introduce this kind of issue when merging Crowdin pull requests as we're lacking a test suite on the project for now.

**Comment by kytwb** [comment]
Shipped in [v5.4.4-beta.2](https://github.com/getferdi/ferdi/releases/tag/v5.4.4-beta.2).
